### PR TITLE
Ex 5 threading workshop 1

### PR DESCRIPTION
**Pull request by MariaKirdun**
Creating thread, runnable, and start it. Handle messages with the handler. Display results in the text view.

### CODE REVIEW

File: app/src/main/java/com/android/academy/fundamentals/workshop_1/WS01ThreadsProblemFragment.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    private fun startThread() {
+        printMessage(getString(R.string.wait))
+        //TODO 5 create your thread and start it
+    }
+
+    private fun startRunnable() {
+        printMessage(getString(R.string.wait))
+        //TODO 8 create your thread and start it
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/workshop_1/WS01ThreadsProblemFragment.kt]: Добавить ранабле тут?

File: app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import com.android.academy.fundamentals.R
+
+class WS01ThreadsSolutionFragment: Fragment(R.layout.fragment_theads) {
+
+    private var threadButton : Button? = null
+    private var threadTextView : TextView? = null
+
+    private val handler = MyHandler()
+
+    private val MESSAGE_KEY = "key"
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt]: const?

File: app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import android.os.*
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import com.android.academy.fundamentals.R
+
+class WS01ThreadsSolutionFragment: Fragment(R.layout.fragment_theads) {
+
+    private var threadButton : Button? = null
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt:12]: Clean views' references on destroy

**Review by emartynov — COMMENTED** [review]
Minor comments and looks super!

File: app/src/main/java/com/android/academy/fundamentals/workshop_1/WS01ThreadsProblemFragment.kt (hunk 1 above)
Comment (emartynov) [reply app/src/main/java/com/android/academy/fundamentals/workshop_1/WS01ThreadsProblemFragment.kt]: B todo текстом

**Review by emartynov — APPROVED** [review]
ЛГТМ

File: app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        val thread = Thread(MyRunnable())
+        thread.start()
+        printMessage(getString(R.string.wait))
+    }
+
+    private fun printMessage(mes: String){
+        threadTextView?.text =  mes
+    }
+
+    inner class MyThread : Thread() {
```
Comment (ermolnik) [inline app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt:52]: qst: is there any point to declare class as inner class? As I remember we didn't explain this point and it is not important for students

**Pull request merged**